### PR TITLE
add Alternative Execution Time functionality

### PR DESCRIPTION
**Pull request by jonatan-alama**
- Added Alternative Execution Time parameter and trigger logic, updated README
- Refactoring of ItemRunner.java
- Refactoring of JenkinsJob.java
- Fixed some misspelled attributes
- Fixed some checkstyle problems

### CODE REVIEW

File: src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/AlternativeExecutionTime.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public static final boolean shouldBuild(Date alternativeExecutionTime, JenkinsJob jenkinsJob, TimeUnit timeUnit)
+    {
+        // timeUnit is hourly, daily
+        // alternativeExecutionTime is set
+        // alternativeExecutionTime has passed
+        if (!TimeUnit.SINGULAR.equals(timeUnit) &&
+            null != alternativeExecutionTime &&
+            alternativeExecutionTime.before(new Date())) {
```
Comment (civitaspo) [inline src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/AlternativeExecutionTime.java]: This conditions are a bit complicated, so it's better to divide these like below.

```
if (TimeUnit.SINGULAR.equals(timeUnit) || alternativeExecutionTime == null) {
    // insufficient requirements
    return false;
}

if (alternativeExecutionTime.after(new Date())) {
    // alternativeExecutionTime has not passed
    return false;
}

...
```

File: src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/JenkinsJob.java [hunk 2]
```diff
@@ -42,31 +40,24 @@ public String url()
...
-            ((AbstractProject) item).scheduleBuild(
-                    0,
-                    buildWithParametersCause(),
-                    new ParametersAction(buildParameters()));
-        }
-        else {
-            item.scheduleBuild(0, defaultCause());
+        if (!(item instanceof AbstractProject)) {
+            RuntimeException e = new RuntimeException(
+                String.format("Error: Not %s, Job: %s", AbstractProject.class.getName(), name()));
```
Comment (civitaspo) [inline src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/JenkinsJob.java]: It's better to output `item.class` for Jenkins operations managers.

File: src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/AlternativeExecutionTime.java (hunk 1 above)
Comment (jonatan-alama) [reply src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/AlternativeExecutionTime.java]: Sure, I agree with you

File: README.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ Jenkins Plugin For [Triglav](https://github.com/sonots/triglav).
 - **Job Message Offset (ReadOnly)**: Job Message Offset which the job consumed. Users cannot configure this value.
 - **Time Zone**: Time Zone of resources.
 - **Time Unit**: Time Unit of resources.
+- **Alternative Execution Time**: Optional time limit at which the Jenkins Project will be build (without parameters from Triglav) if there has been no execution since the last period. Accepts `mm` format when **Time Unit** is `hourly` and `HH:mm` format when **Time Unit** is `daily`. It will be ignored if **Time Unit** is set to `singular` or if the format is not correct. The Project will be built again after the time limit build if Triglav triggers it.
```
Comment (sonots) [inline README.md]: `when the Jenkin Project will be built` ?

Comment (jonatan-alama) [reply README.md]: Thanks for the catch!

File: src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/TimeUnit.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package io.github.triglav_dataflow.jenkins.trigger.polling_triglav;
```
Comment (civitaspo) [inline src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/TimeUnit.java:1]: Reformat this file.

File: src/main/resources/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/PollingTriglavTrigger/config.jelly [hunk 5]
```diff
@@ -33,6 +33,9 @@
   <f:entry title="Time Unit" field="timeUnit">
     <f:select/>
   </f:entry>
+  <f:entry title="Alternative Execution Time" field="alternativeExecutionTime">
+    <f:textbox value="${it.alternativeExecutionTime}" checkUrl="'descriptorByName/io.github.triglav_dataflow.jenkins.trigger.polling_triglav.PollingTriglavTrigger/checkAlternativeExecutionTime?alternativeExecutionTime='+escape(this.value)"/>
```
Comment (civitaspo) [inline src/main/resources/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/PollingTriglavTrigger/config.jelly]: I think `checkUrl` is not required. ref. https://wiki.jenkins.io/display/JENKINS/Basic+guide+to+Jelly+usage+in+Jenkins#BasicguidetoJellyusageinJenkins-Formvalidation

**Review by civitaspo — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/JenkinsJob.java (hunk 2 above)
Comment (jonatan-alama) [reply src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/JenkinsJob.java]: Fixed, thank you

File: src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/TimeUnit.java (hunk 4 above)
Comment (jonatan-alama) [reply src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/TimeUnit.java:1]: Fixed. I will take a look to checkstyle rules and see if we can add a warning for that

File: src/main/resources/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/PollingTriglavTrigger/config.jelly (hunk 5 above)
Comment (jonatan-alama) [reply src/main/resources/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/PollingTriglavTrigger/config.jelly]: Thanks, I checked and its working correctly without it. It was handled by @DataBoundConstructor

**Review by civitaspo — APPROVED** [review]
LGTM!

**Pull request merged**